csharp/Platform.Data.Doublets/FFI/UnitedMemoryLinks.cs

```diff
@@ -0,0 +1,421 @@
+                        var restrictionArray = restrictions.ToArray();
+                        var substitutionArray = substitution.ToArray();
+                        fixed (ulong* restrictionPointer = (ulong[])(object)restrictionArray, substitutionPointer = (ulong[])(object)substitutionArray)
+                        {
+                            return from_u64.Convert(Methods.UInt64UnitedMemoryLinks_Update(_body, restrictionPointer, (nuint)restrictionArray.Length, substitutionPointer, (nuint)substitutionArray.Length));
+                        }
+                    }
+                    default:
+                    {
+                        throw new NotImplementedException();
+                    }
+                }
+            }
+        }
+
+        public void Delete(IList<TLink> restrictions)
+        {
+            unsafe
+            {
+                TLink t = default;
+                switch (t)
+                {
+                    case byte:
+                    {
+                        var restrictionArray = restrictions.ToArray();
+                        fixed (byte* restrictionPointer = (byte[])(object)restrictionArray)
+                        {
+                            return Methods.ByteUnitedMemoryLinks_Delete(_body, restrictionPointer, (nuint)restrictionArray.Length);
+                        }
+                    }
+                    case ushort:
+                    {
+                        var restrictionArray = restrictions.ToArray();
+                        fixed (ushort* restrictionPointer = (ushort[])(object)restrictionArray)
+                        {
+                            return Methods.UInt16UnitedMemoryLinks_Delete(_body, restrictionPointer, (nuint)restrictionArray.Length);
+                        }
+                    }
+                    case uint:
+                    {
+                        var restrictionArray = restrictions.ToArray();
+                        fixed (uint* restrictionPointer = (uint[])(object)restrictionArray)
+                        {
+                            return Methods.UInt32UnitedMemoryLinks_Delete(_body, restrictionPointer, (nuint)restrictionArray.Length);
+                        }
+                    }
+                    case ulong:
+                    {
+                        var restrictionArray = restrictions.ToArray();
+                        fixed (ulong* restrictionPointer = (ulong[])(object)restrictionArray)
+                        {
+                            return Methods.UInt64UnitedMemoryLinks_Delete(_body, restrictionPointer, (nuint)restrictionArray.Length);
+                        }
+                    }
+                    default:
+                    {
+                        throw new NotImplementedException();
+                    }
+                }
+            }
+        }
+
+        protected override void Dispose(bool manual, bool wasDisposed)
+        {
+            if (wasDisposed)
```

Comment:
Don't shoot yourself in the foot
```suggestion
            if (wasDisposed && _body != null)
```